### PR TITLE
planner: add IGNORE_INDEX hint

### PR DESCRIPTION
**Pull request by foreyes**
### What problem does this PR solve? 

add `IGNORE_INDEX` hint and change `INDEX` into `USE_INDEX` hint.

### What is changed and how it works?

Just handle `hintIgnoreIndex` as what `hintIndex` do.

### Check List 

Tests 

 - Unit test

Code changes

 - None

Side effects

 - None

Related changes

 - Need parser change

Release note

 - None

### CODE REVIEW

File: planner/core/logical_plan_builder.go [hunk 1]
```diff
@@ -1972,7 +1974,7 @@ func (b *PlanBuilder) pushTableHints(hints []*ast.TableOptimizerHint, nodeType n
 			preferAggType |= preferHashAgg
 		case HintStreamAgg:
 			preferAggType |= preferStreamAgg
-		case HintIndex:
+		case HintUseIndex:
```
Comment (lzmhhh123) [inline planner/core/logical_plan_builder.go]: What will happen if using both `ignore_index(t, idx_a)` and `use_index(t, idx_a)` in a sql block?

Comment (lzmhhh123) [reply planner/core/logical_plan_builder.go]: Do we need to add a test for it?

Comment (foreyes) [reply planner/core/logical_plan_builder.go]: Seems that we will ignore all indexes... I'll add test and fix it.

Comment (foreyes) [reply planner/core/logical_plan_builder.go]: now it works just like normal index hint. Added some tests, you can check it out.

**Comment by foreyes** [comment]
Add some tests and resolved conflict, PTAL. @lzmhhh123 @lamxTyler

**Pull request merged**